### ISSUE TITLE
[BUG] Exception when starting with host_ip in Docker

### ISSUE THREAD

**Issue by sischnei** [StyraHem/ShellyForHASS#414]
## Environment

- ShellyForHASS release with the issue: 0.1.9
- Last working ShellyForHASS release (if known): -
- Home Assistant Core release with the issue: 0.114.2
- Operating environment (Home Assistant/Supervised/Docker/venv): Docker

## Describe the bug

When starting with `host_ip` option set to my local IP that is used for Home Assistant the following exception is found in the logs:
```
020-09-02 16:00:11 ERROR (MainThread) [homeassistant] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/config/custom_components/shelly/__init__.py", line 187, in start_up
    pys.start()
  File "/usr/local/lib/python3.8/site-packages/pyShelly/__init__.py", line 126, in start
    self._coap.start()
  File "/usr/local/lib/python3.8/site-packages/pyShelly/coap.py", line 28, in start
    self._init_socket()
  File "/usr/local/lib/python3.8/site-packages/pyShelly/coap.py", line 50, in _init_socket
    sock.setsockopt(socket.IPPROTO_IP, socket.IP_ADD_MEMBERSHIP, mreq)
OSError: [Errno 19] No such device
```

Home Assistant starts fine but Shelly integration seems to be not working.

## Steps to Reproduce

Run Home Assistant in Docker with port mappings and configure the host_ip for shelly integration using the local IP of the host (e.g. 192.168.0.2)

## Expected behavior

Home Assistant should start fine and Shelly integration should work, using the public IP of the docker host instead of any docker internal IP.

## Additional context

I'm trying with a Shelly 2.5, the only way I got any kind of integration working thus fur is by using the `discover_by_ip` option. In other issue reports I read that `host_ip` must be configured in Docker environments which kind of makes sense as applications usually do not have access to the "public" IP of the docker host and if that info is required for CoAP then it makes sense. However applying the option is causing issues during startup and thus does not work.

**Comment by david-kalbermatten** [comment]
I'm using a Docker Environment as well and I just gave up on running Home Assistant in `bridge` mode. It saves you a lot of headaches when you switch the Home Assistant Docker to `host` network mode.

Sidenote:
If CoAP and/or mDNS work in your network, you'll be able to use auto-discovery for your Shelly Devices ;D

**Comment by hakana** [comment]
Host-ip is used to force broad cast listener to right interface if multiple network interfaces are available. It must be a ip that is available in the docker container.

**Comment by sischnei** [comment]
Alright - well since I'm using docker in bridge mode, the only available IP is going to be the Docker internal one...
Does ShellyForHASS work properly in bridge-mode using CoAP?
Is there a reason why discover_by_ip disables CoAP (or is this statement not true)?

**Comment by hakana** [comment]
discover_by_ip does not disable CoAP

There is a test script that you can use to test if you can receive CoAP messages in Docker container. 
https://github.com/StyraHem/ShellyForHASS/tree/master/util

You can also use the tcpdump tool to verify you see the CoAP messages in the Docker container.
https://github.com/StyraHem/ShellyForHASS/wiki/FAQ6:-Troubleshoot-CoAP-messages

**Comment by sischnei** [comment]
Thanks for the hint - indeed inside my docker I was not receiving any packets using the test.py script. I'm now running the proxy which, if I understand it correctly, forwards multicast packets to the UDP port opened by HomeAssistant. I had to switch the port mapping, now using port 5684 on the host -> 5683 in the container. I have adapted the proxy.py accordingly.

HomeAssistant now shows "protocols: CoAP-msg, poll" - is that alright?

I have no knowledge of CoAP, how are packets routed back to the Shelly? Directly using the Shelly IP:Port? That's probably we I need to proxy only in the other direction!?
